ffi/capi/src/custom_writeable.rs

```diff
@@ -147,14 +151,25 @@ pub unsafe extern "C" fn icu4x_buffer_writeable(cap: usize) -> *mut ICU4XWriteab
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn icu4x_buffer_writeable_borrow(this: *mut ICU4XWriteable) -> *mut u8 {
+/// Grabs a pointer to the underlying buffer of a writable.
+///
+/// # Safety
+/// - The returned pointer is valid until the passed writable is destroyed.
+/// `this` must be a pointer to a valid [`ICU4XWriteable`] constructed by
+/// [`icu4x_buffer_writeable_create()`].
+pub unsafe extern "C" fn icu4x_buffer_writeable_get_bytes(this: *mut ICU4XWriteable) -> *mut u8 {
```

Comment:
suggestion: `this` can just be `&ICU4XWriteable` here, so no need for the function itself to be unsafe, though of course the returned buffer still has safety requirements.